pytest/tests/mocknet/helpers/neard_runner.py

```diff
@@ -141,6 +143,77 @@ class TestState(Enum):
 }"""
 
 
+class EpochConfigOverrides:
+
+    def __init__(self, epoch_config_overrides):
+        self.common_overrides = ""
+        if 'all' in epoch_config_overrides:
+            self.common_overrides = epoch_config_overrides['all']
+            del epoch_config_overrides['all']
+        self.epoch_config_overrides = epoch_config_overrides
+        self.new_protocol_versions = list(epoch_config_overrides.keys())
+
+    def override_file(self, epoch_config_path, overrides):
+        with open(epoch_config_path, 'r') as f:
+            json_data = json.load(f)
+
+        # Apply the overrides
+        updated_json = jq.compile(overrides).input(json_data).first()
+
+        with open(epoch_config_path, 'w') as f:
+            json.dump(updated_json, f, indent=2)
+
+    def override(self, epoch_config_dir):
+        self._create_missing_epoch_config_files(epoch_config_dir)
+        for protocol_version in self.existing_protocol_versions:
+            protocol_overrides = self.epoch_config_overrides.get(
+                protocol_version, "")
+
+            overrides = []
+            if self.common_overrides != "":
+                overrides.append(self.common_overrides)
+            if protocol_overrides != "":
+                overrides.append(protocol_overrides)
+            overrides = ' | '.join(overrides)
+
+            if overrides == "":
+                continue
+            epoch_config_path = os.path.join(epoch_config_dir,
+                                             f'{protocol_version}.json')
+            logging.info(
+                f'Applying overrides to {epoch_config_path}: {overrides}')
+            self.override_file(epoch_config_path, overrides)
+
+    """
+    Creates the epoch config files before we override them.
+    Non existent epoch config files are created by copying the previous protocol version file.
+    """
+
+    def _create_missing_epoch_config_files(self, epoch_config_dir):
+
+        existing_epoch_config_files = os.listdir(epoch_config_dir)
+        existing_protocol_versions = set([
+            int(re.match(r'(\d+)\.json', f).group(1))
+            for f in existing_epoch_config_files
+        ])
+        previous_protocol_version_file = min(existing_protocol_versions)
+        for protocol_version in sorted(self.new_protocol_versions):
+            if protocol_version in existing_protocol_versions:
```

Comment:
This check will not work because `self.new_protocol_versions` contains strings. Also note that `self.epoch_config_overrides` has string keys. So this needs to all be handled uniformly somehow